Khan/perseus
# [Locked Figure Labels] Add labels to locked lines' defining points in the graph and editor

## Summary:
Add support for locked labels settings within the locked
points that define locked lines.

This allows content authors to directly add labels to
the points on either end of a line/ray/segment instead
of having to create two labels on the line and manually
position them. This also means the data contains more
information about how the labels are being used within
the context of the line.

I also deleted `DefiningPointSettings` so that I wouldn't
have to reimplement all the logic for adding/updating/removing
the locked labels associated with a locked point. Instead, I
consolidated it all into `LockedPointSettings`.

Issue: https://khanacademy.atlassian.net/browse/LEMS-2347

## Test plan:
- Go to http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-labels-flag
- Confirm that neither locked points nor locked lines have visible labels in the editor or the graph
- Go to http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-point-labels-flag
- Confirm that only locked points have visible labels in the editor and the graph
- Play around with the labels and confirm that they look right and update accordingly with the point settings
- Go to http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-line-labels-flag
- Confirm that only the locked lines have visible labels in the editor and graph
- Play around with the labels and confirm that they look right and update accordingly with the line settings

## Screenshots:
| Labels | Points | Lines |
| --- | --- | --- |
| <img width="375" alt="Screenshot 2024-09-18 at 3 28 36 PM" src="https://github.com/user-attachments/assets/85ed6eda-6b03-484e-8bab-7d23c498169c"> | <img width="370" alt="Screenshot 2024-09-18 at 3 28 49 PM" src="https://github.com/user-attachments/assets/59c3d080-bf66-46c6-a11b-2682d5a689c4"> | <img width="381" alt="Screenshot 2024-09-18 at 3 29 22 PM" src="https://github.com/user-attachments/assets/cebc3457-3dc8-43c8-8f53-d05e73657fa8"> |

<img width="912" alt="image" src="https://github.com/user-attachments/assets/6a986bd3-101a-4e3f-af33-aae594ef1af9">
